src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    private final Color[] colors = Color.values();
+    private final int size = colors.length;
```

Comment:
I think it would be a costly operation to create an array from enum's values every time a method is called, more space complexity and longer execution time. What if enum contains thousands of colors? I thought, it would be better to initialize it once.

Reply:
I can agree with you but in your solution you every time create an object  of ``ColorSupplier``` and call the method ```getRandom()``` color only once, so it is the same